README.md

```diff
@@ -28,7 +28,7 @@ This project is no longer maintained by Segment. Instead, [Martin Etmajer](https
 ```bash
 
   Usage:
-    terraform-docs [--no-required] [--no-sort | --sort-inputs-by-required] [--with-aggregate-type-defaults] [json | markdown | md] <path>...
+    terraform-docs [--no-required] [--no-sort | --sort-inputs-by-required] [--with-aggregate-type-defaults] [--markdown-table | --markdown-content] [json | markdown | md] <path>...
```

Comment:
The position of these options should be fixed. Right now, it seems they are global properties that can be applied to any of `json | markdown | md`, whereas they should only be taken into account when the user has selected either `markdown` or `md`. Please see [github.com/docopt/docopt-go](github.com/docopt/docopt-go) for help.

Reply:
The name and position of arguments would then also need to be adapted in other places in `README.md` as well as the documentation header in `main.go`.

Reply:
That was my first approach: `[ json | markdown | md ] [ table | content ]` with correct dependency of subcommand, but I wasn't able to make it work through the `docopt`. To be honest I don't remember what was the issue, but I'll give it another try to see.

Reply:
This might be because we're using an outdated library. There's an issue #82 to upgrade to the official repository. I have this assigned to myself, but feel free to take ownership!

Reply:
There might be a problem with flags, right now I changed to `terraform-docs md table ...` (as opposed to `... md --markdown-table ...`) and it works.

Reply:
Using subcommands is a great choice. Right now, those subcommands would also apply to `json`. Is there a better way of structuring commands / subcommands so that `table` and `document` only apply to `markdown` and `md`?

Reply:
Right, but independent of the documentation, the usage suggests, that these subcommands are available with `json` as well. By the way, I've just pushed #91 to update to the latest revision of `github.com/docopt/docopt.go`.

Reply:
Yes, that's true. I'm not sure what I can do on that regard yet. I'll try to see what can be done.

Reply:
Considering [this](http://try.docopt.org/?doc=Usage%3A%0D%0A++terraform-docs+%5Boptions%5D+json+%3Cpath%3E...%0D%0A++terraform-docs+%5Boptions%5D+%5Bmd+%7C+markdown%5D+%5Btable+%7C+document%5D+%3Cpath%3E...%0D%0A++terraform-docs+-h+%7C+--help%0D%0A%0D%0AOptions%3A%0D%0A++-h%2C+--help+++++++++++++++++++++++show+help+information%0D%0A++--no-required++++++++++++++++++++omit+%22Required%22+column+when+generating+markdown%0D%0A++--no-sort++++++++++++++++++++++++omit+sorted+rendering+of+inputs+and+ouputs%0D%0A++--sort-inputs-by-required++++++++sort+inputs+by+name+and+prints+required+inputs+first%0D%0A++--with-aggregate-type-defaults+++print+default+values+of+aggregate+types%0D%0A++--version++++++++++++++++++++++++print+version%0D%0A%0D%0ATypes+of+markdown+%28optional%29%3A++++++table+will+be+used+if+not+specified%0D%0A++table++++++++++++++++++++++++++++generate+markdown+table+of+document%0D%0A++document+++++++++++++++++++++++++generate+markdown+document+of+document%0D%0A&argv=json+md+..%2Fhome), [this](http://try.docopt.org/?doc=Usage%3A%0D%0A++terraform-docs+%5Boptions%5D+json+%3Cpath%3E...%0D%0A++terraform-docs+%5Boptions%5D+%5Bmd+%7C+markdown%5D+%5Btable+%7C+document%5D+%3Cpath%3E...%0D%0A++terraform-docs+-h+%7C+--help%0D%0A%0D%0AOptions%3A%0D%0A++-h%2C+--help+++++++++++++++++++++++show+help+information%0D%0A++--no-required++++++++++++++++++++omit+%22Required%22+column+when+generating+markdown%0D%0A++--no-sort++++++++++++++++++++++++omit+sorted+rendering+of+inputs+and+ouputs%0D%0A++--sort-inputs-by-required++++++++sort+inputs+by+name+and+prints+required+inputs+first%0D%0A++--with-aggregate-type-defaults+++print+default+values+of+aggregate+types%0D%0A++--version++++++++++++++++++++++++print+version%0D%0A%0D%0ATypes+of+markdown+%28optional%29%3A++++++table+will+be+used+if+not+specified%0D%0A++table++++++++++++++++++++++++++++generate+markdown+table+of+document%0D%0A++document+++++++++++++++++++++++++generate+markdown+document+of+document%0D%0A&argv=md+json+..%2Fhome) and [this](http://try.docopt.org/?doc=Usage%3A%0D%0A++terraform-docs+%5Boptions%5D+json+%3Cpath%3E...%0D%0A++terraform-docs+%5Boptions%5D+%5Bmd+%7C+markdown%5D+%5Btable+%7C+document%5D+%3Cpath%3E...%0D%0A++terraform-docs+-h+%7C+--help%0D%0A%0D%0AOptions%3A%0D%0A++-h%2C+--help+++++++++++++++++++++++show+help+information%0D%0A++--no-required++++++++++++++++++++omit+%22Required%22+column+when+generating+markdown%0D%0A++--no-sort++++++++++++++++++++++++omit+sorted+rendering+of+inputs+and+ouputs%0D%0A++--sort-inputs-by-required++++++++sort+inputs+by+name+and+prints+required+inputs+first%0D%0A++--with-aggregate-type-defaults+++print+default+values+of+aggregate+types%0D%0A++--version++++++++++++++++++++++++print+version%0D%0A%0D%0ATypes+of+markdown+%28optional%29%3A++++++table+will+be+used+if+not+specified%0D%0A++table++++++++++++++++++++++++++++generate+markdown+table+of+document%0D%0A++document+++++++++++++++++++++++++generate+markdown+document+of+document%0D%0A&argv=md+table+..%2Fhome) I think we won't be able to only depends on docopt and we need to do some internal sanity check too. The same argument can be said for mutually exclusive `--no-sort` and `--sort-inputs-by-required`. So the Usage can be shorten to something like in the example links above.

And even considering the upcoming `v1.0.0` release even some of those options can be even shortened (`--aggregate`, etc.)